pathd/path_zebra.c

```diff
@@ -284,6 +526,70 @@ static void path_zebra_label_manager_connect(void)
 	}
 }
 
+static void path_nht_update(struct prefix *nh, int nhnum)
+{
+	struct srte_policy *policy;
+	struct prefix sid_srv6 = {};
+	struct srte_candidate *candidate;
+	struct srte_segment_list *segment_list;
+
+	RB_FOREACH (policy, srte_policy_head, &srte_policies) {
+		if (policy->endpoint.ipa_type != AF_INET6)
+			continue;
+
+		candidate = policy->best_candidate;
+		if (!candidate)
+			continue;
+
+		segment_list = candidate->lsp->segment_list;
+		if (!segment_list)
+			continue;
+
+		/* srv6 segment lists are registered */
+		if (!CHECK_FLAG(segment_list->flags,
+				F_SEGMENT_LIST_NHT_REGISTERED))
+			continue;
+
+		if (!path_zebra_nht_get_srv6_prefix(segment_list, &sid_srv6))
+			continue;
+		if (!IPV6_ADDR_SAME(&sid_srv6.u.prefix6, &nh->u.prefix6))
+			continue;
+		if (nhnum)
+			path_zebra_add_sr_policy(policy, segment_list);
+		else
+			path_zebra_delete_sr_policy(policy);
+	}
+}
+
+static int path_zebra_nexthop_update(ZAPI_CALLBACK_ARGS)
```

Comment:
Please use the `zclient->nexthop_update` call back function instead.  I don't want to remember to have to change pathd when I add/remove data the encode/decode of nexthops

Reply:
thanks